File: 2102-find-the-middle-index-in-array/find-the-middle-index-in-array.py
class Solution(object):
    def findMiddleIndex(self, nums):
        
        n = len(nums)
        prefix = 0

        for i in range(n):
            prefix += nums[i]

        left_sum = 0
        
        for i in range(n):
            right_sum = prefix - nums[i] - left_sum

            if left_sum == right_sum:
                return i

            left_sum += nums[i]

        return -1

File: 2102-find-the-middle-index-in-array/find-the-middle-index-in-array.java
class Solution {
    public int findMiddleIndex(int[] nums) {

        int n = nums.length;

        if (n == 1)
            return 0;

        int prefix = 0;

        for (int i = 0; i < n; i++)
            prefix += nums[i];

        int left_sum = 0;

        for (int i = 0; i < n; i++) {
            int right_sum = prefix - nums[i] - left_sum;

            if (right_sum == left_sum)
                return i;

            left_sum += nums[i];
        }

        return -1;
    }
}